examples/chat/proto/chat.proto

```diff
@@ -21,47 +21,51 @@ package oak.examples.chat;
 import "google/protobuf/empty.proto";
 
 message CreateRoomRequest {
-  // Name of the room; this is only used for UI purposes (and so does not need
-  // to be unique).
+  // Name of the room; this is only used for UI purposes (and so does not need to be unique).
   string name = 1;
-  // ID used to identify the room; knowledge of this value allows entry to the
-  // room.  The client should create a random bearer token of sufficient
-  // entropy to use as the ID.
+
+  // ID used to identify the room; knowledge of this value allows entry to the room. The client
+  // should create a random bearer token of sufficient entropy to use as the room ID.
+  //
+  // TODO(#212): Once more of the labelling / policy system is implemented in the Oak Runtime, the
+  // secrecy of each chat room will not be based on the secrecy of this value, rather by the labels
+  // attached to individual messages by the various clients.
   bytes room_id = 2;
-  // ID used to administer the room; knowledge of this value allows control over
-  // the room (including the ability to destroy the room). The client should
-  // create a random bearer token of sufficient entropy to use as the ID.)
-  bytes admin_id = 3;
+
+  // Bearer token used to administer the room; knowledge of this value allows control over the room
+  // (including the ability to destroy the room). The client should create a random bearer token of
+  // sufficient entropy to use as the admin token.
+  bytes admin_token = 3;
 }
 
 message DestroyRoomRequest {
-  bytes admin_id = 1;
+  bytes room_id = 1;
+  bytes admin_token = 2;
 }
 
-message JoinRoomRequest {
+message SubscribeRequest {
   bytes room_id = 1;
-  // Handle for the user; this is only used for UI purposes (and so does not
-  // need to be unique nor persistent).
-  string user_handle = 2;
```

Comment:
Removing this removes the ability for the client UI to emit messages like "Oak Discussion: tiziano88 has joined the room", so again seems to be a reduction in functionality in exchange for a more minimal API.

(Again, modulo the fact that user names are entirely spoofable like room names are.)

Reply:
I think eventually the streaming API must be complemented by a REST API that lets a user retrieve all messages (perhaps with a limit) for a room that they have permission to access. What controls the set of recipients of a chat room should (eventually) be the policies associated with each message, not the set of clients with a currently open gRPC connection. The streaming API may be a nice optimisation to allow receiving new messages more quickly, but I don't think it can be the only (or even main) way of interacting with the app. 

In fact perhaps the API may be:

```protobuf
service Chat {
  // Create a new room with the provided fixed set of participants.
  rpc CreateRoom(CreateRoomRequest) returns (CreateRoomResponse);
  // Retrieve rooms that the current user has access to. Note this does not actually retrieve the messages in the rooms, just their metadata, including the list of participants (which is public).
  rpc GetRooms(GetRoomsRequest) returns (GetRoomsResponse);
  // Retrieve all messages for the specified room. It requires the client to have permission to read such messages, via some authentication mechanism (e.g. public key, OpenID connect token)
  rpc GetMessages(GetMessagesRequest) returns (GetMessagesResponse);
  // Send a message to the specified room. The client must attach a label to the message corresponding to the list of participants with which the room was created.
  rpc SendMessage(SendMessageRequest) returns (SendMessageResponse);
}
```

WDYT?

Reply:
/cc @ipetr0v as he has been looking into implementing the actual Android UI for the hypothetical chat app

Reply:
Note: I am not suggesting we change the API in this PR, but wanted to give a heads-up on what I think the API may look like in order for it to be used by a realistic looking Android app that we can put together as a demo.

@daviddrysdale as we discussed yesterday, maybe the API should keep the `Subscribe` method in addition to `GetMessages`, since polling would be inconvenient in practice, and also it's nice to demonstrate streaming capabilities of the application.

Reply:
Adding more functionality (like `GetMessages`) seems sensible, in line with making this closer to a real chat app, to confirm that such a thing works fine in Oak.  

I guess there's a tension between that, and making the example app as simple as possible so it can be used as a testbed for exploring how the policy system might work. 

The latter is very much in line with what the example has been used for so far, as a vehicle for confirming that various parts of the runtime (dynamic node creation, persistent server-side streaming, multi-client parallel use, multi-Application parallel use) work OK.

So I guess removing room names and joining announcements is fine given that aim, but at some point we might have to reinstate equivalent functionality.

Reply:
Thanks for your thoughts, I'll follow up with another PR to discuss the exact API that we may use for a "real" chat app.